md2conf/application.py

```diff
@@ -15,16 +23,70 @@ def update_document(
     api.update_page(document.page_id, document.xhtml())
 
 
+
+def synchronize_pages(api: ConfluenceSession, path: str, options: ConfluenceDocumentOptions) -> None:
+    if os.path.isdir(path):
+        synchronize_directory(api, path, options)
+    elif os.path.isfile(path):
+        synchronize_page(api, path, options)
+    else:
+        raise ValueError(f"expected: valid file of directory path. got: {path}")
+
+
 def synchronize_page(
-    api: ConfluenceSession, path: str, options: ConfluenceDocumentOptions
+    api: ConfluenceSession, path: str, options: ConfluenceDocumentOptions, page_metadata: dict[str, str] = dict()
 ) -> None:
     page_path = os.path.abspath(path)
     base_path = os.path.dirname(page_path)
 
-    document = ConfluenceDocument(path, options)
+    LOGGER.info(f"synchronize_page: {page_path}")
+    document = ConfluenceDocument(path, options, page_metadata)
 
     if document.space_key:
         with api.switch_space(document.space_key):
             update_document(api, document, base_path)
     else:
         update_document(api, document, base_path)
+
+
+def synchronize_directory(api: ConfluenceSession, dir: str, options: ConfluenceDocumentOptions) -> None:
+    page_metadata = dict()
+    LOGGER.info(f"synchronize_directory: {dir}")
+
+    # Step 1: build index of all page metadata
+    for root, directories, files in os.walk(dir):
+        for filename in files:
+            # Extract the file extension from the path
+            file_extension = os.path.splitext(filename)[1]
+            abs_path = os.path.abspath(os.path.join(os.path.abspath(root), filename))
```

Comment:
I am not sure I fully understand the purpose of the outer call to `os.path.abspath`, looks like we don't need it at all.

Reply:
You are correct. fixed!